src/Api/WebhookApi.php

```diff
@@ -4,35 +4,53 @@
 
 namespace Sylius\PayPalPlugin\Api;
 
-use GuzzleHttp\ClientInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
 
 final class WebhookApi implements WebhookApiInterface
 {
     private ClientInterface $client;
 
+    private RequestFactoryInterface $requestFactory;
+
+    private StreamFactoryInterface $streamFactory;
+
     private string $baseUrl;
 
-    public function __construct(ClientInterface $client, string $baseUrl)
-    {
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
```

Comment:
```suggestion
        private readonly RequestFactoryInterface $requestFactory,
```
Do we want to maintain backward compatibility?

Reply:
changed